src/core/control/tools/EditSelection.cpp

```diff
@@ -99,6 +99,23 @@ EditSelection::EditSelection(UndoRedoHandler* undo, const vector<Element*>& elem
     view->rerenderPage();
 }
 
+EditSelection::EditSelection(UndoRedoHandler* undo, const vector<Element*>& elements, XojPageView* view,
+                             const PageRef& page, Layer* layer):
+        snappingHandler(view->getXournal()->getControl()->getSettings()) {
```

Comment:
The vector `elements` must be the layer's content (otherwise, you get a memory leak).

```suggestion
EditSelection::EditSelection(UndoRedoHandler* undo, XojPageView* view,
                             const PageRef& page, Layer* layer):
        snappingHandler(view->getXournal()->getControl()->getSettings()) {
    const auto& elements = layer->getElements();
```